src/main/java/taxi/web/filter/AuthenticationFilter.java

```diff
@@ -0,0 +1,45 @@
+package taxi.web.filter;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Set;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.annotation.WebFilter;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+
+@WebFilter(urlPatterns = "/*")
+public class AuthenticationFilter implements Filter {
+    private final Set<String> allowedUrls = new HashSet<>();
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+        allowedUrls.add("/login");
+        allowedUrls.add("/drivers/add");
+    }
+
+    @Override
+    public void doFilter(ServletRequest req, ServletResponse resp, FilterChain filterChain)
+            throws IOException, ServletException {
+        HttpServletRequest request = (HttpServletRequest) req;
+        HttpServletResponse response = (HttpServletResponse) resp;
+        HttpSession session = request.getSession();
+        Long driverId = (Long) session.getAttribute("driverId");
+        if (driverId == null
+                && allowedUrls.contains(request.getServletPath())) {
+            filterChain.doFilter(request, response);
+            return;
+        }
+        if (driverId == null) {
+            response.sendRedirect(request.getContextPath() + "/login");
+            return;
+        }
```

Comment:
![image](https://user-images.githubusercontent.com/117590814/214566044-990a4c3a-931e-4eba-99d3-4e4bd756e61b.png)
once again, trust me, it's possible 😉